File: 백준/스택-수열.py
import sys

n = int(sys.stdin.readline())

stack, result = list(), list()
count = 1

for _ in range(n):
    data = int(sys.stdin.readline())

    while count <= data:
        stack.append(count)
        result.append('+')
        count += 1
    if data == stack[-1]:
        stack.pop()
        result.append('-')
    else:
        sys.stdout.write('NO')
        exit(0)

sys.stdout.write('\n'.join(result))

File: 백준/스택수열.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const loop = input.shift();
const stack = [];
let answer = [];
let count = 1;

for (let i = 0; i < loop; i++) {
  const number = parseInt(input[i], 10);

  while (count <= number) {
    stack.push(count);
    answer.push('+');
    count++;
  }

  if (stack[stack.length - 1] !== number) {
    answer = ['NO'];
    break;
  }

  answer.push('-');
  stack.pop();
}

console.log(answer.join('\n'));
